package.json

```diff
@@ -45,6 +45,7 @@
     "@testing-library/jest-dom": "5.11.0",
     "@testing-library/react": "10.4.6",
     "@testing-library/react-hooks": "3.3.0",
+    "@testing-library/user-event": "^12.0.11",
```

Comment:
What is the requirement for this library? We currently use `fireEvent.click` for our interaction testing.

If required we should lock the dependency range as well.

Reply:
The `user-event` library isn't strictly required. I think it could removed, but here my rationale: its APIs simulate more accurate user interaction than manually using a single `fireEvent` API. For example, to assert against hovering, `user.hover(element)` does a lot more than simply `fireEvent.mouseEnter(element)`. It fires other events that would occur in an actual hover:

```jsx
function hover(element, init) {
  if (isLabelWithInternallyDisabledControl(element)) return

  fireEvent.pointerOver(element, init)
  fireEvent.pointerEnter(element, init)
  if (!element.disabled) {
    fireEvent.mouseOver(element, getMouseEventOptions('mouseover', init))
    fireEvent.mouseEnter(element, getMouseEventOptions('mouseenter', init))
  }
  fireEvent.pointerMove(element, init)
  if (!element.disabled) {
    fireEvent.mouseMove(element, getMouseEventOptions('mousemove', init))
  }
}
```

The same applies to the two other APIs I wanted to use in this PR: [`user.click(element)`](https://github.com/testing-library/user-event/blob/master/src/click.js#L81) and [`user.tab(element)`](https://github.com/testing-library/user-event/blob/master/src/tab.js#L22). Also, the `user-event` package is included in the default `create-react-app` now. Looks like it's getting wide adoption 🎉 .

_Note: I forgot to use `user.hover(element)` in my [original PR](https://github.com/zendeskgarden/react-components/pull/812/files#diff-a0cecbd65bdd8164d6cb5cdc9acf81c5R117)._

Reply:
We could slowly migrate to moving towards using `user-events` APIs over-time. An introduction of this package could include a wider team discussion. So I leave this package out of this PR.